discretisedfield/field.py

```diff
@@ -3479,3 +3480,210 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (10, 10, 10)
+        >>> cell = (1, 1, 1)
+        >>> mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+        >>> field = df.Field(mesh=mesh, dim=3, value=(1, 0, 0), norm=1.)
+        ...
+        >>> field
+        Field(...)
+
+        2. Create `xarray.DataArray` from field
+
+        >>> xa = field.to_xarray()
+        >>> xa
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10, comp: 3)>
+        ...
+        Coordinates:
+          * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * comp     (comp) <U1 'x' 'y' 'z'
+        Attributes:
+            units:    None
+
+        3. Select values of `x` component
+
+        >>> xa.sel(comp='x')
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10)>
+        ...
+        Coordinates:
+          * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+            comp     <U1 'x'
+        Attributes:
+            units:    None
+
+        """
+        if not isinstance(name, str):
+            msg = "Name argument must be a string."
+            raise TypeError(msg)
+
+        if units is not None and not isinstance(units, str):
+            msg = "Units argument must be a string."
+            raise TypeError(msg)
+
+        axes = ['x', 'y', 'z']
+
+        data_array_coords = {
+            axis: np.fromiter(self.mesh.axis_points(axis), dtype=float)
+            for axis in axes
+        }
+
+        if 'unit' in self.mesh.attributes:
+            geo_units_dict = dict.fromkeys(axes, self.mesh.attributes['unit'])
+        else:
+            geo_units_dict = dict.fromkeys(axes, 'm')
+
+        if self.dim > 1:
+            data_array_dims = axes + ['comp']
+            if self.components is not None:
+                data_array_coords['comp'] = self.components
+            field_array = self.array
+        else:
+            data_array_dims = axes
+            field_array = np.squeeze(self.array, axis=-1)
+
+        data_array = xr.DataArray(field_array,
+                                  dims=data_array_dims,
+                                  coords=data_array_coords,
+                                  name=name,
+                                  attrs=dict(units=units))
+
+        for dim in geo_units_dict:
+            data_array[dim].attrs['units'] = geo_units_dict[dim]
+
+        return data_array
+
+    @classmethod
+    def from_xarray(cls, xa):
+        if not isinstance(xa, xr.DataArray):
+            msg = "Argument must be a xr.DataArray."
+            raise TypeError(msg)
+
+        if xa.ndim not in [3, 4]:
+            msg = "DataArray dimensions must be 3 or 4."
```

Comment:
Is 3 for scalar and 4 for vector field? This error message might be confusing to users. (what is a 4D field)

Reply:
Yes, the dimensions of the input `DataArray` are `x`, `y`, `z` (corresponding to the geometry), and `comp` (corresponding to the field). If we have the scalar field, than `comp` is not present and we have just the tree dimensions corresponding to the geometry. If we have a vector field (could be n dimensional), than `comp` is present which will hold the components of the field. Hence, I wanted to check that we have either three or four dimensions for scalar field or n dimensional vector field respectively.

I agree that the message is a bit opaque. Maybe we can output, "DataArray dimensions should be 3 for a scalar field and 4 for a vector field."?